src/main/java/org/broadinstitute/hellbender/tools/walkers/varianteval/VariantEval.java

```diff
@@ -0,0 +1,838 @@
+            throw new CommandLineException.BadArgumentValue("ST", "Variants cannot be stratified by sample and family at the same time");
+
+        if (perFamilyIsEnabled && sampleDB.getTrios().isEmpty())
+            throw new CommandLineException.BadArgumentValue("ST", "Cannot stratify by family without *.ped file");
+
+
+        if ( intervalsFile != null ) {
+            boolean fail = true;
+            for ( final VariantStratifier vs : stratificationObjects ) {
+                if ( vs.getClass().equals(IntervalStratification.class) )
+                    fail = false;
+            }
+            if ( fail )
+                throw new CommandLineException.BadArgumentValue("ST", "stratIntervals argument provided but -ST IntervalStratification not provided");
+        }
+
+        // Initialize the evaluation contexts
+        createStratificationStates(stratificationObjects, evaluationClasses);
+
+        // Load ancestral alignments
+        if (ancestralAlignmentsFile != null) {
+            try {
+                ancestralAlignments = new IndexedFastaSequenceFile(ancestralAlignmentsFile);
+            } catch (FileNotFoundException e) {
+                throw new GATKException(String.format("The ancestral alignments file, '%s', could not be found", ancestralAlignmentsFile.getAbsolutePath()));
+            }
+        }
+    }
+
+    private void checkForIncompatibleEvaluatorsAndStratifiers( final List<VariantStratifier> stratificationObjects,
+                                                             Set<Class<? extends VariantEvaluator>> evaluationClasses) {
+        for ( final VariantStratifier vs : stratificationObjects ) {
+            for ( Class<? extends VariantEvaluator> ec : evaluationClasses )
+                if ( vs.getIncompatibleEvaluators().contains(ec) )
+                    throw new CommandLineException.BadArgumentValue("ST and ET",
+                            "The selected stratification " + vs.getName() + 
+                                    " and evaluator " + ec.getSimpleName() +
+                                    " are incompatible due to combinatorial memory requirements." +
+                                    " Please disable one");
+        }
+    }
+
+    final void createStratificationStates(final List<VariantStratifier> stratificationObjects, final Set<Class<? extends VariantEvaluator>> evaluationObjects) {
+        final List<VariantStratifier> strats = new ArrayList<VariantStratifier>(stratificationObjects);
+        stratManager = new StratificationManager<>(strats);
+
+        logger.info("Creating " + stratManager.size() + " combinatorial stratification states");
+        for ( int i = 0; i < stratManager.size(); i++ ) {
+            EvaluationContext ec = new EvaluationContext(this, evaluationObjects);
+            stratManager.set(i, ec);
+        }
+    }
+
+    private class PositionAggregator {
+        private String contig = null;
+        private int start;
+        private int end;
```

Comment:
sure, but any concern about unnecessarily creating more SimpleInterval objects than needed if we keep expanding the interval?  That's going to be comparatively rare, but could happen with indels or when running w/ multiple VCF inputs.

Reply:
No - SimpleInterval should be fine.

Reply:
As you might have seen I already added that anyway.  however, for my own edification what exactly does using SimpleInterval gain here?

Reply:
Clarity, simplicity, correctness, economy of expression.... The class is caching an interval, which is what SimpleInterval is for, and it was already using SimpleInterval in one place to update the read and feature contexts, but storing the individual components. Now, instead of two representations across four variables, the class has one representation and one variable. Even though the code is fairly simple to start with, using SimpleInterval is way, way better all around. In my opinion.

Also, I didn't notice this until I saw the diff for this change, but because the code was able to update `contig`, `start`, and `end` independently, they were getting out of sync before - `end` wasn't updated when the new interval exceeded the end of the old one.

Reply:
alright, good points.